addons/mrp/models/stock_move.py

```diff
@@ -335,7 +335,8 @@ def _compute_display_assign_serial(self):
     def _onchange_product_uom_qty(self):
         if self.raw_material_production_id and self.has_tracking == 'none':
             mo = self.raw_material_production_id
-            new_qty = float_round((mo.qty_producing - mo.qty_produced) * self.unit_factor, precision_rounding=self.product_uom.rounding)
+            uom_rounding = self.product_uom.rounding or self.product_id.uom_id.rounding
```

Comment:
But then, once the user sets an UoM, the onchange will not be triggered and `quantity` won't be correct
I would rather add the UoM field in the onchange dependencies and just skip the onchange if we don't have any value for the field